### PR TITLE
Integration test fixes

### PR DESCRIPTION
**Pull request by phoet**
there are a couple of changes that make `rake test:integration` run locally without failing tests. hopefully that also fixes the tests on travis.

### CODE REVIEW

**Comment by phoet** [comment]
@karmi did you ever find out why the percolate queries fail on travis? is that a problem of versions? 

the changes i made allow the local test-suite to pass with ruby 2 and elasticsearch 0.90.2

**Comment by karmi** [comment]
@phoet It works when you wipe _percolator index, but that can't be done normally, people might lose data. Will try to get to all the issues soon, limited online now...

File: test/integration/active_record_searchable_test.rb [hunk 1]
```diff
@@ -613,7 +613,7 @@ module ::Rails; end
       context "percolated search" do
         setup do
           delete_registered_queries
-          delete_percolator_index if ENV['TRAVIS']
+          delete_percolator_index
```
Comment (karmi) [inline test/integration/active_record_searchable_test.rb:616]: @phoet I'd rather have failing tests, then wipe the percolator index for people :) Someone might just run the Tire suite, not realizing we're wiping the index, he might have some hard-to-get-again data there... Let's be rather more careful here. Sorry about the silence about the pull requests, I'm at a vacation now, will try to squeeze some review & merging into next week.

Comment (phoet) [inline test/integration/active_record_searchable_test.rb:616]: hey, no problem!

i think that the tests must not fail locally. is there a way to seperate
these destructive tests somehow and have them run only on travis?

Am Donnerstag, 18. Juli 2013 schrieb Karel Minarik :

Comment (karmi) [inline test/integration/active_record_searchable_test.rb:616]: Yeah, that was my thinking in adding the `if ENV['TRAVIS']`, might be actually better to run percolator tests with this "flag". Obviously, the best solution would be to launch a separate ES service just for tests -- have something on backburner for that :)

**Comment by karmi** [comment]
@phoet Thanks for looking into this! In the end, I've changed the way we run "destructive" tests without introducing the special context, it might be more transparent this way. Hopefully Travis will get green again :)

**Pull request closed without merging**